main/mySpace/session_admin_teachers.php

```diff
@@ -0,0 +1,292 @@
+<?php
+/* For licensing terms, see /license.txt */
+/**
+ * Teacher report.
+ *
+ * @package chamilo.reporting
+ */
+$cidReset = true;
+
+require_once __DIR__.'/../inc/global.inc.php';
+
+api_block_anonymous_users();
+
+$allowToTrack = api_is_platform_admin(true, true) || api_is_teacher();
+
+if (!$allowToTrack) {
+    api_not_allowed(true);
+}
+
+$export_csv = isset($_GET['export']) && $_GET['export'] == 'csv' ? true : false;
+$keyword = isset($_GET['keyword']) ? Security::remove_XSS($_GET['keyword']) : null;
+$nameTools = get_lang('Teachers');
+$this_section = SECTION_TRACKING;
+
+$interbreadcrumb[] = ["url" => "index.php", "name" => get_lang('MySpace')];
+
+if (isset($_GET["user_id"]) && $_GET["user_id"] != "" && !isset($_GET["type"])) {
+    $interbreadcrumb[] = ["url" => "teachers.php", "name" => get_lang('Teachers')];
+}
+
+if (isset($_GET["user_id"]) && $_GET["user_id"] != "" && isset($_GET["type"]) && $_GET["type"] == "coach") {
+    $interbreadcrumb[] = ["url" => "coaches.php", "name" => get_lang('Tutors')];
+}
+
+function get_count_users()
+{
+    $keyword = isset($_GET['keyword']) ? Security::remove_XSS($_GET['keyword']) : null;
+    $sessions = SessionManager::getSessionsForAdmin(api_get_user_id(), [], false, [], 'all');
+    $coachesList = [];
+    $students = [];
+
+    foreach ($sessions as $session) {
+        $sessionId = $session['id'];
+        $courseList = SessionManager::getCoursesInSession($sessionId);
+        $courses = [];
+        if (!empty($courseList)) {
+            foreach ($courseList as $courseId) {
+                $coaches = CourseManager::get_coachs_from_course($sessionId, $courseId);
+                foreach ($coaches as $coach) {
+                    if (!empty($keyword)) {
+                        if (stripos($coach['full_name'], $keyword) === false) {
+                            continue;
+                        }
+                    }
+
+                    if (!in_array($coach['user_id'], $coachesList)) {
+                        $coachesList[] = $coach['user_id'];
+                        $students[] = $coach;
+                    }
+                }
+            }
+        }
+    }
+
+    $count = count($students);
+
+    return $count;
+}
+
+function get_users($from, $limit, $column, $direction)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $all_datas is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5201882)